openpype/hosts/unreal/api/pipeline.py

```diff
@@ -60,6 +62,23 @@ def show_tools_dialog(self):
 
         show_tools_dialog()
 
+    def update_context_data(self, data, changes):
+        unreal.log_warning("update_context_data")
+        unreal.log_warning(data)
+        content_path = unreal.Paths.project_content_dir()
+        op_ctx = content_path + CONTEXT_CONTAINER
+        with open(op_ctx, "w+") as f:
```

Comment:
I am thinking that this might cause potential issues with concurrency - imagine multiple artists working on the same project, updating the context file and reading it at the same time. I bet that this will result in `IOError`.

The concurrency issue in this case really scares me and we need to think about solution.